extensions/data-workspace/src/dialogs/newProjectDialog.ts

```diff
@@ -64,14 +75,20 @@ export class NewProjectDialog extends DialogBase {
 		}
 	}
 
+	override onCancelButtonClicked(): void {
+		this.newProjectDialogComplete?.resolve();
+		this.dispose();
+	}
+
 	override async onComplete(): Promise<void> {
 		try {
 
 			TelemetryReporter.createActionEvent(TelemetryViews.NewProjectDialog, TelemetryActions.NewProjectDialogCompleted)
 				.withAdditionalProperties({ projectFileExtension: this.model.projectFileExtension, projectTemplateId: this.model.projectTypeId })
 				.send();
 
-			await this.workspaceService.createProject(this.model.name, vscode.Uri.file(this.model.location), this.model.projectTypeId, this.model.targetPlatform);
+			this.projectUri = await this.workspaceService.createProject(this.model.name, vscode.Uri.file(this.model.location), this.model.projectTypeId, this.model.targetPlatform);
+			this.newProjectDialogComplete?.resolve();
```

Comment:
will it wait for newProjectDialogComplete.resolve?

Reply:
Yup the `openSpecificProjectNewProjectDialog()` opens the dialog and then waits for this to resolve and returns the new project uri.
https://github.com/microsoft/azuredatastudio/blob/kisantia/filterTargetVersion/extensions/data-workspace/src/dialogs/newProjectDialog.ts#L30